streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -522,6 +523,12 @@ private void restoreChangelog(final ChangelogMetadata changelogMetadata) {
             // do not trigger restore listener if we are processing standby tasks
             if (changelogMetadata.stateManager.taskType() == Task.TaskType.ACTIVE) {
                 try {
+                    // first trigger the store's specific listener if its registered callback is also an lister,
+                    // then trigger the user registered global listener
+                    final StateRestoreCallback restoreCallback = changelogMetadata.storeMetadata.restoreCallback();
```

Comment:
Nit: we get `storeMetadata` above already (similar below) also the `if` is redundant  -- should we unify it?